@AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/engine/fn_engineController.sqf

```diff
@@ -109,32 +119,34 @@ if (   ((_eng1State == "OFF" && _eng2State == "OFF") || (_eng1PwrLvrState in ["O
 };
 //End Autorotation handler
 
-private _maxTQ    = getNumber (_config >> "engMaxTQ");
-private _limitTQ  = 0.0;
-private _limitRPM = getNumber (_config >> "engIdleNP");
-
-private _realRPM = [_heli] call fza_sfmplus_fnc_getRtrRPM;
+if (_flightModel != "sfmplus") then {
+    private _maxTQ    = getNumber (_config >> "engMaxTQ");
+    private _limitTQ  = 0.0;
+    private _limitRPM = getNumber (_config >> "engIdleNP");
 
-if (_isSingleEng) then {
-    _limitTQ = _heli getVariable "fza_sfmplus_maxTQ_SE";
-} else {
-    _limitTQ = _heli getVariable "fza_sfmplus_maxTQ_DE";
-};
-
-private _droopVal = (_rtrRPM - _limitRPM) / (_maxTQ - _limitTQ);
-private _droopRPM = _rtrRPM - ((_engPctTQ - _limitTQ) * _droopVal);
-_droopRPM = [_droopRPM, _limitRPM, _rtrRPM] call BIS_fnc_clamp;
-
-if (_heli getHitPointDamage "hithrotor" == 1.0) exitWith {};
+    private _realRPM = [_heli] call fza_sfmplus_fnc_getRtrRPM;
 
-private _lastUpdate = _heli getVariable ["fza_sfmplus_lastUpdate", 0];
-if (cba_missionTime > _lastUpdate + MIN_TIME_BETWEEN_UPDATES) then {
-    _rtrRPM = _droopRPM;
-        if (_realRPM >= _rtrRPM || (_eng1State == "OFF" && _eng2State == "OFF" && !_isAutorotating)) then {
-        _heli setHitpointDamage ["hithrotor", 0.9];
+    if (_isSingleEng) then {
+        _limitTQ = _heli getVariable "fza_sfmplus_maxTQ_SE";
     } else {
-        _heli setHitpointDamage ["hithrotor", 0.0];
-        _heli engineOn true;
+        _limitTQ = _heli getVariable "fza_sfmplus_maxTQ_DE";
+    };
+
+    private _droopVal = (_rtrRPM - _limitRPM) / (_maxTQ - _limitTQ);
+    private _droopRPM = _rtrRPM - ((_engPctTQ - _limitTQ) * _droopVal);
+    _droopRPM = [_droopRPM, _limitRPM, _rtrRPM] call BIS_fnc_clamp;
+
+    if (_heli getHitPointDamage "hithrotor" == 1.0) exitWith {};
+
+    private _lastUpdate = _heli getVariable ["fza_sfmplus_lastUpdate", 0];
+    if (cba_missionTime > _lastUpdate + MIN_TIME_BETWEEN_UPDATES && _rtrRPM > 0.05) then {
+        _rtrRPM = _droopRPM;
+        if (_realRPM > _rtrRPM) then {
```

Comment:
it would still be best to keep
` _rtrRPM > 0.05`
removed as it would allow sfm+ users to fly just by holding shift

Reply:
That shouldn't be the reason for it, and I know it definitely needs to be in there. The shift thing is likely caused by a logic issue further up the chain.